Add docs for Controller::middleware() (#7029)
cakephp/docs: en/appendices/4-3-migration-guide.rst

New Features
============

Controller
----------

> - ``Controller::middleware()`` was added. Enabling you to define middleware that
>   is only applied before each action in a single controller. See
>   :ref:`controller-middleware` for more information.

Reviewer: ```suggestion
- ``Controller::middleware()`` was added which allows you to define middleware 
  for a single controller only. See :ref:`controller-middleware` for more information.
```
